processor/stream/processor.go

```diff
@@ -142,32 +146,34 @@ func RUMProcessor(cfg *config.Config, tcfg *transform.Config) *Processor {
 				modelDecoder: metricset.DecodeEvent,
 			},
 			"error": {
-				er.ModelSchema(),
-				er.DecodeEvent,
+				schema:       er.ModelSchema(),
+				modelDecoder: er.DecodeEvent,
 			},
 		},
+		metadataSchema: metadata.ModelSchema(),
 	}
 }
 
 func RUMV3Processor(cfg *config.Config, tcfg *transform.Config) *Processor {
 	return &Processor{
 		Tconfig:      *tcfg,
-		Mconfig:      model.Config{Experimental: cfg.Mode == config.ModeExperimental},
+		Mconfig:      model.Config{Experimental: cfg.Mode == config.ModeExperimental, HasShortFieldNames: true},
 		MaxEventSize: cfg.MaxEventSize,
 		models: map[string]processorModel{
-			"transaction": {
+			"x": {
 				schema:       transaction.RUMV3Schema,
 				modelDecoder: transaction.DecodeRUMV3Event,
 			},
-			"span": {
+			"y": {
 				schema:       span.RUMV3Schema,
 				modelDecoder: span.DecodeRUMV3Event,
 			},
-			"error": {
+			"e": {
 				schema:       er.RUMV3Schema,
 				modelDecoder: er.DecodeRUMV3Event,
 			},
 		},
+		metadataSchema: metadata.RUMV3ModelSchema(),
```

Comment:
The `metricset` model is missing here, I'll file a ticket for that to follow up in a separate PR